server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateDataStreamService.java

```diff
@@ -179,10 +179,11 @@ public static void validateTimestampFieldMapping(String timestampFieldName, Mapp
 
         Map<String, Object> parsedTemplateMapping =
             MapperService.parseMapping(NamedXContentRegistry.EMPTY, mapperService.documentMapper().mappingSource().string());
-        String configuredPath = ObjectPath.eval("_doc._data_stream_timestamp.path", parsedTemplateMapping);
-        if (timestampFieldName.equals(configuredPath) == false) {
-            throw new IllegalArgumentException("[_data_stream_timestamp] meta field doesn't point to data stream timestamp field [" +
-                timestampFieldName + "]");
+        Boolean enabled = ObjectPath.eval("_doc._data_stream_timestamp.enabled", parsedTemplateMapping);
```

Comment:
For me, it'd be nice to just remove this whole method `validateTimestampMapping`. It's a little fragile to be checking the raw JSON. And I'm not sure it adds much more confidence (users should really not be messing around with the internal `_data_stream_timestamp` field). I don't feel strongly about this though.

Reply:
I don't think it is too fragile. If for some reason the json format changes then tests fail and this method needs to be adjusted.

I still prefer to keep this method for in the case for some unknown reason the `_data_stream_timestamp` field mapping doesn't get applied. There is no mechanism in place that yells when this meta field mapper isn't applied when creating a backing index of data stream.

Reply:
This works for me. I approved the PR earlier and feel it's ready to merge.